From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexandre Derumier <aderumier@odiso.com>
Date: Mon, 14 May 2018 17:06:52 +0200
Subject: [PATCH] manual interfaces : set link up

by default ifdown2 don't set link up for manual interfaces.

classic ifupdown set link up for manual interfaces.
I think this has change with jessie and systemd.

https://github.com/CumulusNetworks/ifupdown2/issues/4
---
 ifupdown2/ifupdown/ifupdownmain.py | 15 +--------------
 1 file changed, 1 insertion(+), 14 deletions(-)

diff --git a/ifupdown2/ifupdown/ifupdownmain.py b/ifupdown2/ifupdown/ifupdownmain.py
index 45391ee..2daf49f 100644
--- a/ifupdown2/ifupdown/ifupdownmain.py
+++ b/ifupdown2/ifupdown/ifupdownmain.py
@@ -97,11 +97,6 @@ class ifupdownMain(ifupdownBase):
             (ifaceobj.link_privflags & ifaceLinkPrivFlags.VRF_SLAVE)):
             self._keep_link_down(ifaceobj)
             return
-        # if not a logical interface and addr method is manual,
-        # ignore link admin state changes
-        if (ifaceobj.addr_method == 'manual' and
-            not ifaceobj.link_kind):
-            return
         if self._delay_admin_state:
             self._delay_admin_state_iface_queue.append(ifaceobj.name)
             return
@@ -134,11 +129,6 @@ class ifupdownMain(ifupdownBase):
         # there is no real interface behind it
         if ifaceobj.type == ifaceType.BRIDGE_VLAN:
             return
-        # if not a logical interface and addr method is manual,
-        # ignore link admin state changes
-        if (ifaceobj.addr_method == 'manual' and
-            not ifaceobj.link_kind):
-            return
         if self._delay_admin_state:
             self._delay_admin_state_iface_queue.append(ifaceobj.name)
             return
@@ -527,10 +517,7 @@ class ifupdownMain(ifupdownBase):
     def _set_iface_role_n_kind(self, ifaceobj, upperifaceobj):
 
         # If addr_method is set and link is not a logical interface,
-        # set flag KEEP_LINK_DOWN. addr_method == 'manual' only applies to
-        # logical interfaces.
-        if (ifaceobj.addr_method == 'manual' and not ifaceobj.link_kind):
-            ifaceobj.link_privflags |= ifaceLinkPrivFlags.KEEP_LINK_DOWN
+        # set flag KEEP_LINK_DOWN.
 
         if (upperifaceobj.link_kind & ifaceLinkKind.BOND):
             self._set_iface_role(ifaceobj, ifaceRole.SLAVE, upperifaceobj)
-- 
2.11.0

